src/main/java/net/fabricmc/loom/task/AbstractRunTask.java

```diff
@@ -135,7 +151,49 @@ public void exec() {
 		setWorkingDir(new File(getProjectDir().get(), getInternalRunDir().get()));
 		environment(getInternalEnvironmentVars().get());
 
-		super.exec();
+		// Wrap with XVFB if enabled and on Linux
+		if (shouldUseXvfb()) {
+			LOGGER.info("Using XVFB for headless client execution");
+			execWithXvfb();
+		} else {
+			super.exec();
+		}
+	}
+
+	private boolean shouldUseXvfb() {
+		return getUseXvfb().get();
+	}
+
+	private void execWithXvfb() {
+		String xvfbRunPath = "/usr/bin/xvfb-run";
+
+		// Get the java executable path - try toolchain first, fallback to system property
+		String javaExec;
+
+		if (getJavaLauncher().isPresent()) {
+			javaExec = getJavaLauncher().get().getExecutablePath().getAsFile().getAbsolutePath();
+		} else {
+			// Fallback: construct path from JAVA_HOME
+			String javaHome = System.getProperty("java.home");
+			javaExec = javaHome + File.separator + "bin" + File.separator + "java";
+		}
```

Comment:
https://github.com/FabricMC/fabric-loom/blob/e1cda212bd053da4299fbb8ed4312dbe2832a7f9/src/main/java/net/fabricmc/loom/task/prod/ClientProductionRunTask.java#L126C20-L126C63

Can this just be `getJavaLauncher().get().getExecutablePath()`?